modules/core/src/main/java/org/locationtech/jts/geom/impl/CoordinateArraySequence.java

```diff
@@ -153,7 +201,12 @@ public Coordinate getCoordinateCopy(int i) {
   public void getCoordinate(int index, Coordinate coord) {
```

Comment:
As a sanity check, this seems like an easy place for a user to do something silly.  If they pass in the wrong kind of Coordinate, they could get a no-Z/M-support error message, right?  

In JTS 2.0, it'd be great to see if we can be clever enough to handle this at the type-system.

Reply:
In this case we trust the `coord` set methods to handle things appropriately.